File: pack/effects/offset/effect.js
function effect(x, y, pixel) {
    x += offset;
    y += offset;
    if(x >= size.x + jump.x) {
        x -= size.x;
    }
    if(y >= size.y + jump.y) {
        y -= size.y;
    }
    return ImageAt(x, y);
}

File: pack/effects/offset/effect.lua
function effect(pixel)
    local x = pixel["x"] + offset_x
    local y = pixel["y"] + offset_y

    while x >= img_size["x"] + jump_x do
        x = x - img_size["x"]
    end
    while y >= img_size["x"] + jump_x do
        y = y - img_size["x"]
    end

    return at(x, y)
end